### PR TITLE
Convert ha-progress-button to lit

### PR DESCRIPTION
**Pull request by ludeeus**
## Proposed change

:man_shrugging: :point_up: 

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/components/buttons/ha-progress-button.ts [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+class HaProgressButton extends LitElement {
+  @property({ type: Boolean }) public disabled = false;
+
+  @property({ type: Boolean }) public progress = false;
+
+  @query("#container") private _buttonContainer!: HTMLDivElement;
+
+  public render(): TemplateResult | void {
+    return html`
+      <div id="container">
```
Comment (zsarnett) [inline src/components/buttons/ha-progress-button.ts:31]: Do we need this container? Should be able to set attributes on the element itself

File: src/components/buttons/ha-progress-button.ts [hunk 2]
```diff
@@ -24,31 +23,27 @@ class HaProgressButton extends LitElement {
 
   @property({ type: Boolean }) public progress = false;
 
-  @query("#container") private _buttonContainer!: HTMLDivElement;
-
   public render(): TemplateResult | void {
```
Comment (zsarnett) [inline src/components/buttons/ha-progress-button.ts:26]:
```suggestion
  public render(): TemplateResult {
```

File: src/components/buttons/ha-progress-button.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+  progress: boolean;
+}
+
+@customElement("ha-progress-button")
+class HaProgressButton extends LitElement {
+  @property({ type: Boolean }) public disabled = false;
+
+  @property({ type: Boolean }) public progress = false;
+
+  public render(): TemplateResult | void {
```
Comment (zsarnett) [inline src/components/buttons/ha-progress-button.ts:26]:
```suggestion
  public render(): TemplateResult  {
```

File: src/components/buttons/ha-progress-button.ts [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    this._tempClass("success");
+  }
+
+  public actionError(): void {
+    this._tempClass("error");
+  }
+
+  protected firstUpdated(changedProps: PropertyValues) {
+    super.firstUpdated(changedProps);
+    this.addEventListener("click", (ev) => this._buttonTapped(ev));
```
Comment (zsarnett) [inline src/components/buttons/ha-progress-button.ts:52]: I dont think we need this

**Review by zsarnett — APPROVED** [review]
I hate that the element exposes functions that set the classes instead of a property that's updated. This is how it was created and changing it is probably out of the PR. Maybe it is the best way. idk.

Looks good 👍

File: src/components/buttons/ha-progress-button.ts [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+  PropertyValues,
+  TemplateResult,
+} from "lit-element";
+
+import "../ha-circular-progress";
+
+export interface HaProgressButtonElement extends HTMLElement {
+  disabled: boolean;
+  progress: boolean;
+}
```
Comment (bramkragten) [inline src/components/buttons/ha-progress-button.ts:18]: Why have this?

File: src/components/buttons/ha-progress-button.ts [hunk 6]
```diff
@@ -0,0 +1,103 @@
+import "@material/mwc-button";
+import {
+  css,
+  CSSResult,
+  customElement,
+  html,
+  LitElement,
+  property,
+  PropertyValues,
```
Comment (bramkragten) [inline src/components/buttons/ha-progress-button.ts:9]:
```suggestion
```

**Review by bramkragten — APPROVED** [review]

**Pull request merged**